src/main/java/com/teamabnormals/abnormals_core/core/AbnormalsCore.java

```diff
@@ -84,12 +86,17 @@ public AbnormalsCore() {
 			}
 		});
 
-		DistExecutor.runWhenOn(Dist.CLIENT, () -> () -> {
-			((IReloadableResourceManager) Minecraft.getInstance().getResourceManager()).addReloadListener(ENDIMATION_DATA_MANAGER);
-			modEventBus.addListener(this::clientSetup);
-		});
+		DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> modEventBus.addListener(EventPriority.NORMAL, false, ColorHandlerEvent.Block.class, event ->
+		{
+			IResourceManager resourceManager = Minecraft.getInstance().getResourceManager();
+			if (resourceManager instanceof IReloadableResourceManager)
+			{
+				((IReloadableResourceManager) resourceManager).addReloadListener(ENDIMATION_DATA_MANAGER);
+			}
+		}));
 
 		modEventBus.addListener(EventPriority.LOWEST, this::commonSetup);
+		modEventBus.addListener(this::clientSetup);
```

Comment:
Moving this outside of running client-side only now causes a crash. Which makes me infer that you didn't test it on servers?